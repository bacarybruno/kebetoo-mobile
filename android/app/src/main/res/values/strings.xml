<resources>
    <string name="app_name">Kebetoo</string>
</resources>
