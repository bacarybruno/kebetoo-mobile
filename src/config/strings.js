import LocalizedStrings from 'react-native-localization';
import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';
import 'dayjs/locale/en';
import 'dayjs/locale/fr';

import frStrings from '@assets/locales/fr';
import enStrings from '@assets/locales/en';

const strings = new LocalizedStrings({
  fr: frStrings,
  en: enStrings,
});

dayjs.locale(strings.getLanguage());
dayjs.extend(relativeTime);

export const updateAppLocale = (locale) => {
  dayjs.locale(locale);
  strings.setLanguage(locale);
};

export default strings;
