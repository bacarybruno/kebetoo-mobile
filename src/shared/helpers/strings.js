export const capitalize = (s) => {
  if (typeof s !== 'string') return ''
  const splitCharacters = '. '
  const parts = s.split(splitCharacters)
  return parts
    .map((s) => s.charAt(0).toUpperCase() + s.slice(1).toLowerCase())
    .join(splitCharacters)
}
