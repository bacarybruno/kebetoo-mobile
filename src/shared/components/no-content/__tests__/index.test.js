import React from 'react'

import setupTest from '@app/config/jest-setup'
import strings from '@app/config/strings'

import NoContent from '../index'
import Typography, { types } from '../../typography'


const givenNoContent = setupTest(NoContent)({
  title: strings.general.no_content,
  text: strings.comments.no_content,
  children: <Typography type={types.body} text="child" />,
})

it('renders NoContent', () => {
  const { wrapper } = givenNoContent()
  expect(wrapper.toJSON()).toMatchSnapshot()
})
