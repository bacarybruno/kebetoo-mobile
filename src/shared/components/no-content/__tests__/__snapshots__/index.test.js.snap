// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NoContent 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#000000",
          "fontFamily": "System",
          "fontSize": 20,
          "fontWeight": "400",
          "letterSpacing": 0.361328,
          "lineHeight": 25,
        },
        false,
        false,
        false,
        Array [
          Object {
            "color": "rgba(0, 122, 255, 255)",
          },
          Object {
            "backgroundColor": "transparent",
            "fontFamily": "System",
            "fontWeight": "600",
          },
          undefined,
        ],
      ]
    }
  >
    No content yet
  </Text>
  <Text
    style={
      Array [
        false,
        Object {
          "backgroundColor": "transparent",
          "color": "#000000",
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "600",
          "letterSpacing": -0.408,
          "lineHeight": 22,
        },
        false,
        false,
        Array [
          Object {
            "color": "rgba(0,0,0,0.87)",
          },
          Object {
            "backgroundColor": "transparent",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          undefined,
        ],
      ]
    }
  >
    Be the first to add a comment! ðŸ‘‡
  </Text>
  <Text
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#000000",
          "fontFamily": "System",
          "fontSize": 15,
          "fontWeight": "400",
          "letterSpacing": -0.24,
          "lineHeight": 20,
        },
        Array [
          Object {
            "color": "rgba(0,0,0,0.87)",
          },
          Object {
            "backgroundColor": "transparent",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          undefined,
        ],
      ]
    }
  >
    child
  </Text>
</View>
`;
